tests/snapshots_test.go

```diff
@@ -0,0 +1,358 @@
+package snapshots_test
+
+import (
+	"io"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/hexops/autogold/v2"
+	"github.com/sanity-io/litter"
+)
+
+func Function(string, int) (string, error) {
+	return "", nil
+}
+
+type BlankStruct struct{}
+
+type BasicStruct struct {
+	Public  int
+	private int
+}
+
+type IntAlias int
+
+type InterfaceStruct struct {
+	Ifc interface{}
+}
+
+type RecursiveStruct struct {
+	Ptr *RecursiveStruct
+}
+
+type CustomMap map[string]int
+
+type CustomMultiLineDumper struct {
+	Dummy int
+}
+
+func (cmld *CustomMultiLineDumper) LitterDump(w io.Writer) {
+	_, _ = w.Write([]byte("{\n  multi\n  line\n}"))
+}
+
+type CustomSingleLineDumper int
+
+func (csld CustomSingleLineDumper) LitterDump(w io.Writer) {
+	_, _ = w.Write([]byte("<custom>"))
+}
+
+type unexportedStruct struct {
+	x int
+}
+
+type StructWithUnexportedType struct {
+	unexported unexportedStruct
+}
+
+func TestLitterSdump(t *testing.T) {
+	standardCfg := litter.Options{}
+
+	t.Run("primitives work", func(t *testing.T) {
+		messages := make(chan string, 3)
+		sends := make(chan<- int64, 1)
+		receives := make(<-chan uint64)
+
+		val := []interface{}{
+			false,
+			true,
+			7,
+			int8(10),
+			int16(10),
+			int32(10),
+			int64(10),
+			uint8(10),
+			uint16(10),
+			uint32(10),
+			uint64(10),
+			uint(10),
+			float32(12.3),
+			float64(12.3),
+			float32(1.0),
+			float64(1.0),
+			complex64(12 + 10.5i),
+			complex128(-1.2 - 0.1i),
+			(func(v int) *int { return &v })(10),
+			"string with \"quote\"",
```

Comment:
You could use


```suggestion
			`string with "quote"`,
```

Reply:
These are lifted directly from the old test, but sure I guess